tex/generic/pgf/graphdrawing/lua/pgf/gd/trees/doc.lua

```diff
@@ -293,11 +293,9 @@ example
   };
 ]]
 
---[[
-% TODOsp: codeexamples: the next example needs the library `arrows.meta`
---]]
 example
 [[
+\usetikzlibrary{shapes.misc, arrows.meta, decorations.pathmorphing}
```

Comment:
To make settings like this line colored in gray in the pgfmanual, in normal examples they are passed to the `preamble` option, like
```tex
\begin{codeexample}[preamble={\usetikzlibrary{...}]
...
\end{codeexample}
```

But here this example is processed by Lua function `example`
https://github.com/pgf-tikz/pgf/blob/12f17834c849a659b8829bbc9122bc459f585917/tex/generic/pgf/graphdrawing/lua/pgf/gd/trees/doc.lua#L16

which may not support equivalence of `preamble` option yet.

Also there are more such `TODOsp`s:
```
$ rg --no-heading -F '% TODOsp:' -g '*.lua'
tex/generic/pgf/graphdrawing/lua/pgf/gd/control/doc.lua:102:% TODOsp: codeexamples: the following 3 examples need these libraries
tex/generic/pgf/graphdrawing/lua/pgf/gd/control/doc.lua:171:% TODOsp: codeexamples: the following 4 examples need these libraries
tex/generic/pgf/graphdrawing/lua/pgf/gd/trees/doc.lua:297:% TODOsp: codeexamples: the next example needs the library `arrows.meta`
```

All these `TODOsp`s were added by @Mo-Gul in fa4e75714894696a45c91fd23b7739c0d71b2116 so he/she may know better than me.

Reply:
Hi @muzimuzhi thanks for the comments.

However, it appears he specification of `\usetikzlibrary` can be done within the `document` environment i.e. it doesn't need to be in the preamble.

In other words, the proposed changes should work as is.  See below for a full working tex file, which generates the expected graph:
```latex
\documentclass{article}
\usepackage{tikz}
\begin{document}
\usetikzlibrary {graphs,graphdrawing} \usegdlibrary {trees}
\usetikzlibrary{shapes.misc, arrows.meta, decorations.pathmorphing}
\tikz \graph [binary tree layout] {
  Knuth -> {
    Beeton -> Kellermann [second] -> Carnes,
    Tobin -> Plass -> { Lamport, Spivak }
  }
};\qquad
\tikz [>={Stealth[round,sep]}]
  \graph [binary tree layout, grow'=right, level sep=1.5em,
          nodes={right, fill=blue!50, text=white, chamfered rectangle},
                  edges={decorate,decoration={snake, post length=5pt}}]
  {
    Knuth -> {
      Beeton -> Kellermann [second] -> Carnes,
      Tobin -> Plass -> { Lamport, Spivak }
    }
  };
\end{document}
```
What am I missing?

Reply:
What I worried was that the way your added `\usetikzlibrary{...}` line won't make it being highlighted in gray in the typeset pgfmanual. The `preamble` option is different from the preamble part of a tex document.

Turns out I was wrong, it is correctly highlighted, see [the `pgfmanual` artifact](https://github.com/pgf-tikz/pgf/actions/runs/9323599439?pr=1335) from GitHub Actions workflow run.

![image](https://github.com/pgf-tikz/pgf/assets/6376638/d9f9986d-af3f-45f3-afc3-858bcd43f489)

Reply:
Probably this "work out of expectation" behavior is caused by a flaw in macros for pgfmanual, see #957.

Learnt from 6a0e08db272f09091ffdfc3f8766c91979138750, Lua examples also support options now. Didn't check whether the `example` function supports it or not.
https://github.com/pgf-tikz/pgf/blob/12f17834c849a659b8829bbc9122bc459f585917/tex/generic/pgf/graphdrawing/lua/pgf/gd/doc.lua#L111-L121

Reply:
I think I've found a better solution to this.  Will post an update shortly.  (Apparently `l3build doc -q` takes about 8.5 minutes on my mac.)

Reply:
@muzimuzhi Could you please approve the pending workflow, let me know if there are any more concerns or additional improvements this PR can make?  Thanks.

Reply:
The workflow also runs in your fork if you enable GitHub Actions in it.
https://github.com/hansonchar/pgf/actions

Reply:
Thanks.  The workflows completed successfully.  The artifacts can be found at https://github.com/hansonchar/pgf/actions/runs/9396877417

Reply:
> What I worried was that the way your added `\usetikzlibrary{...}` line won't make it being highlighted in gray in the typeset pgfmanual. The `preamble` option is different from the preamble part of a tex document.
> 
> Turns out I was wrong, it is correctly highlighted, see [the `pgfmanual` artifact](https://github.com/pgf-tikz/pgf/actions/runs/9323599439?pr=1335) from GitHub Actions workflow run.
> 
> ![image](https://private-user-images.githubusercontent.com/6376638/336631575-d9f9986d-af3f-45f3-afc3-858bcd43f489.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTc3NTEzODYsIm5iZiI6MTcxNzc1MTA4NiwicGF0aCI6Ii82Mzc2NjM4LzMzNjYzMTU3NS1kOWY5OTg2ZC1hZjNmLTQ1ZjMtYWZjMy04NThiY2Q0M2Y0ODkucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDYwNyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA2MDdUMDkwNDQ2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OWMwYTZjMWZhZDQ5ZjhkYjM3ZGQwZGIzMWQwMWNlMDMzOWQ4MmM2NTg2ODI1Y2EwNDcxZjgyODRhMzc0MmU2ZSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.5ylCWsupK0nyh9Woe3vfhnCq3fjUxhV9X31pLKcqNXo)

While you're at it, could you double check the spelling of the name "Kellerman(n)"? [Barbara Beeton noticed](https://chat.stackexchange.com/transcript/message/65762914#65762914) that there might be one `n` too much (see also https://tug.org/mail-archives/twg-tds/1994-October/000000.html if that's the person from your graph).

Reply:
I've done some investigation, and it appears that
1. "Kellermann" with two 'n's is a common German surname. Examples include people like the German author [Bernhard Kellermann](https://en.wikipedia.org/wiki/Bernhard_Kellermann).
1. "Kellerman" with one 'n' is also a recognized surname, used by people like the American author [Jonathan Kellerman](https://en.wikipedia.org/wiki/Jonathan_Kellerman).

In this repository, "Kellermann" appears to refer to "Bernhard Kellermann".  If so, two 'n's should be the correct spelling.

Reply:
@hansonchar Somebody who died in 1951 is related to TeX?

Reply:
The book "Der Tunnel" (by Kellermann) was typeset using TeX by Knuth himself, and later became a test case for LTeX, as Beeton used it to demonstrate the capabilities of LaTeX.

Reply:
Based on your links, it appears "Kellermann" refers to the contributor "David Kellerman", and if so it should be a single 'n'.

I'll make the change shortly.

Reply:
FYI: Changes made, and the [workflow](https://github.com/hansonchar/pgf/actions/runs/9420609224) successfully completed.